src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerNoticeOfChange.java

```diff
@@ -106,35 +106,82 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
         final CaseDetails<CaseData, State> details,
         final CaseDetails<CaseData, State> beforeDetails
     ) {
-        log.info("Caseworker notice of change aboutToSubmit callback started for Case Id: {}", details.getId());
+        log.info("About to start submitting Notice of Change");
 
         final var data = details.getData();
-        final var applicant = data.getNoticeOfChange().getWhichApplicant() == APPLICANT_1
-            ? data.getApplicant1()
-            : data.getApplicant2();
+        final var beforeData = beforeDetails.getData();
 
+        final boolean isApplicant1 = data.getNoticeOfChange().getWhichApplicant() == APPLICANT_1;
+        final var orgPolicyCaseAssignedRole = isApplicant1 ? APPLICANT_1_SOLICITOR : APPLICANT_2_SOLICITOR;
+
+        final var applicant = isApplicant1 ? data.getApplicant1() : data.getApplicant2();
+        final var beforeApplicant = isApplicant1 ? beforeData.getApplicant1() : beforeData.getApplicant2();
+
+        updateSolicitorInformation(data, orgPolicyCaseAssignedRole, applicant);
+
+        final var roles = data.getNoticeOfChange().getWhichApplicant() == APPLICANT_1
+            ? List.of(CREATOR.getRole(), APPLICANT_1_SOLICITOR.getRole())
+            : List.of(APPLICANT_2.getRole(), APPLICANT_2_SOLICITOR.getRole());
+
+        NoticeType noticeType = calculateNoticeType(applicant, beforeApplicant);
+
+        switch (noticeType) {
```

Comment:
You could use polymorphism to replace a switch statement here by moving the logic to the enum: https://www.baeldung.com/a-guide-to-java-enums